src/components/page-active-component/external-dns.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+  Accordion,
+  Button,
+  Checkbox,
+  TextField,
+  Typography,
+} from '@equinor/eds-core-react';
+import * as PropTypes from 'prop-types';
+import {
+  ChangeEvent,
+  FunctionComponent,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  ExternalDns,
+  useGetEnvironmentQuery,
+  useUpdateComponentExternalDnsTlsMutation,
+} from '../../store/radix-api';
+import { ExternalDnsAliasHelp } from '../external-dns-alias-help';
+import { Alert } from '../alert';
+import { ScrimPopup } from '../scrim-popup';
+import { errorToast, successToast } from '../global-top-nav/styled-toaster';
+import { getFetchErrorData, getFetchErrorMessage } from '../../store/utils';
+import { ExternalDNSList } from '../external-dns';
+
+type TlsFormData = {
+  certificate?: string;
+  privateKey?: string;
+};
+
+const TLSFieldsForm: FunctionComponent<{
+  disabled?: boolean;
+  onChange?: (tlsData: TlsFormData) => void;
+}> = ({ disabled, onChange }) => {
+  const [tlsData, setTlsData] = useState<TlsFormData>({});
+  useEffect(() => onChange?.(tlsData), [onChange, tlsData]);
+
+  return (
+    <fieldset className="grid grid--gap-large" disabled={disabled}>
+      <TextField
+        id="certString"
+        label="Certificate"
+        multiline
+        rows={5}
+        onChange={(ev: ChangeEvent<HTMLInputElement>) =>
+          setTlsData((v) => ({ ...v, certificate: ev.target.value }))
+        }
+      />
+      <TextField
+        id="privateKeyString"
+        label="Private Key"
+        multiline
+        rows={5}
+        onChange={(ev: ChangeEvent<HTMLInputElement>) =>
+          setTlsData((v) => ({ ...v, privateKey: ev.target.value }))
+        }
+      />
+    </fieldset>
+  );
+};
+
+const EditTLSForm: FunctionComponent<{
+  appName: string;
+  envName: string;
+  componentName: string;
+  fqdn: string;
+  onSaveSuccess?: () => void;
+  onSaveError?: (error) => void;
+}> = ({
+  appName,
+  envName,
+  componentName,
+  fqdn,
+  onSaveSuccess,
+  onSaveError,
+}) => {
+  const [{ certificate, privateKey }, setTlsData] = useState<TlsFormData>({});
+  const tlsDataIsValid = useMemo(
+    () => certificate?.length > 0 && privateKey?.length > 0,
+    [certificate, privateKey]
+  );
+  const [skipValidation, setSkipValidation] = useState(false);
+  const [
+    saveFunc,
+    { isLoading: isSaving, isError: isSaveError, error: saveError },
+  ] = useUpdateComponentExternalDnsTlsMutation();
+  const saveApiError = useMemo(
+    () => (isSaveError ? getFetchErrorData(saveError) : null),
+    [isSaveError, saveError]
+  );
+
+  return (
+    <form
+      onSubmit={async (ev) => {
+        ev.preventDefault();
+        try {
```

Comment:
move the function out of the html and to a desginated function... consider using `handlePromiseWithToast` helper

Reply:
fixed